model/flow/entity.go

```diff
@@ -24,3 +27,21 @@ func EntitiesToIDs[T Entity](entities []T) []Identifier {
 	}
 	return ids
 }
+
+// Deduplicate entities in a slice by the ID method
+func Deduplicate[T IDEntity](entities []T) []T {
```

Comment:
Is it a requirement of this function to keep the ordering of `entities`? If not it could simply be

```
set := make(map[Identifier]T, len(entities))
for _, entity := range entities {
	set[entity.ID()] = entity
}
return maps.Values(set)
```

Reply:
Yes, it simplifies the code a bit, the memory usage is similar as it still creates a map, however the ordering will be different from the original, this creates additional case that the caller has to worry about. I think I would prefer the caller not having to worry about the additional case. 

I will mention in the comment that the original ordering is preserved.